core/src/main/java/org/apache/spark/metrics/prometheus/client/exporter/TextFormatWithTimestamp.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.metrics.prometheus.client.exporter;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Enumeration;
+
+import io.prometheus.client.Collector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TextFormatWithTimestamp {
+    private static final Logger logger = LoggerFactory.getLogger(TextFormatWithTimestamp.class);
+
+    /**
+     * Content-type for text version 0.0.4.
+     */
+    public static final String CONTENT_TYPE_004 = "text/plain; version=0.0.4; charset=utf-8";
+
+    private static StringBuilder jsonMessageLogBuilder = new StringBuilder();
+
+    public static void write004(Writer writer,
+                                Enumeration<Collector.MetricFamilySamples> mfs)throws IOException {
+        write004(writer, mfs, null);
+    }
+
+    /**
+     * Write out the text version 0.0.4 of the given MetricFamilySamples.
+     */
+    public static void write004(Writer writer,Enumeration<Collector.MetricFamilySamples> mfs,
+                                String timestamp) throws IOException {
+    /* See http://prometheus.io/docs/instrumenting/exposition_formats/
+     * for the output format specification. */
+    while(mfs.hasMoreElements()) {
+        Collector.MetricFamilySamples metricFamilySamples = mfs.nextElement();
```

Comment:
I think `for(Collector.MetricFamilySamples s: Collections.list(mfs)) {`  would be nicer.

Reply:
Also, method body is not indented well.

Reply:
This class has been refactored.